plugin/service/host/plugin.go

```diff
@@ -42,24 +40,19 @@ var _ pb.HostPluginServiceServer = (*HostPlugin)(nil)
 func (p *HostPlugin) OnCreateCatalog(ctx context.Context, req *pb.OnCreateCatalogRequest) (*pb.OnCreateCatalogResponse, error) {
 	catalog := req.GetCatalog()
 	if catalog == nil {
-		return nil, status.Error(codes.InvalidArgument, "catalog is nil")
-	}
-
-	secrets := catalog.GetSecrets()
-	if secrets == nil {
-		return nil, status.Error(codes.InvalidArgument, "secrets are required")
+		return nil, ierrors.BadRequestStatus("catalog is required")
 	}
 
 	attrs := catalog.GetAttributes()
 	if attrs == nil {
-		return nil, status.Error(codes.InvalidArgument, "attributes are required")
+		return nil, ierrors.BadRequestStatus("attributes are required")
```

Comment:
nit: I am not loving the `ierrors`. I wonder if we can remove the need for std `errors` By moving the error.As checks into an internal function in the errors package. But not critical.

Reply:
I tracked the use of std `errors` to a single `errors.New` call [here](https://github.com/hashicorp/boundary-plugin-aws/blob/main/plugin/service/host/plugin.go#L629). I turned that into a `fmt.Errorf` call and was able to remove the `ierrors` alias - https://github.com/hashicorp/boundary-plugin-aws/pull/49/commits/e6a35ac2852f47d2c8b101efaaa33df75f9f8e4c